### PR TITLE
Take the namespace of generated files from the RootNamespace property

### PR DESCRIPTION
**Pull request by justinwritescode**
Change to take the namespace from the RootNamespace property instead of from the AssemblyName.  This makes the generated code configurable.

### CODE REVIEW

**Comment by StefH** [comment]
@justinwritescode 
Did you use a different newline for the files? Because it seems that all lines are changed?

**Comment by justinwritescode** [comment]
I wrote it on a Mac....sorry....

**Comment by justinwritescode** [comment]
Ah...yes...it must be the line endings.  When I click "Hide whitespace changes" it shows up fine. Sorry about that....

**Comment by StefH** [comment]
Can you edit this on a windows pc ? Or change the line-endings to CRLF?

**Comment by justinwritescode** [comment]
Actually, it looks like your line endings are inconsistent.  Have a look at IFileGenerator.cs.  On the left is the repo code, right is a version I just opened and saved with CRLF endings.

image

And here's FluentBuilderClassesGenerator.IDictionary.cs:

image

And here's Header.cs:

image

I checked some of the others I didn't touch and it looks like your files pretty much alternate CRLF and LF line endings....

Anyway, these all have consistent CRLF line endings now.  If it's too confusing for you to look at, you can click the gear icon in the diff viewer and click "Hide whitespace" as shown below.

image

Cheers! And thanks for a great lib!

**Comment by StefH** [comment]
I will take a look tomorrow.

Thank you for investigating.

File: src/FluentBuilderGenerator/Wrappers/IGeneratorExecutionContextWrapper.cs [hunk 1]
```diff
@@ -9,13 +9,14 @@ internal interface IGeneratorExecutionContextWrapper
 {
     /// <see cref="Compilation.AssemblyName"/>
     public string AssemblyName { get; }
+    public string NamespaceName { get; }
```
Comment (StefH) [inline src/FluentBuilderGenerator/Wrappers/IGeneratorExecutionContextWrapper.cs:12]: Can you add some comment for this new property?

**Review by StefH — CHANGES_REQUESTED** [review]
1. Can you write some comments in your PR on how that "build_property.RootNamespace" should be used? (Then I will copy to this to the readme)
2. And can you add a unit-test for this change? (Maybe you need a new test-project for this?)

**Comment by StefH** [comment]
@justinwritescode 
Did you have time yet  to check my review comments?

**Comment by justinwritescode** [comment]
Yes but I haven't had time to implement yet...

**Pull request merged**